tricity/tricity/spiders/trojmiasto.py

```diff
@@ -1,20 +1,50 @@
 import scrapy
 from tricity.items import HouseItem
+import re
+import json
+import hashlib
+
 
 class HousesspiderSpider(scrapy.Spider):
     name = 'tricity'
     allowed_domains = ['ogloszenia.trojmiasto.pl']
     start_urls = ['https://ogloszenia.trojmiasto.pl/nieruchomosci/dom/gdynia/']
 
+    
+    
     def parse(self, response):
-        houses = response.xpath("//div[@id = 'ogl__list__wrap']").get()
-
+        houses = response.css('#ogl__list__wrap .list__item')
+        
         for house in houses:
             house_item = HouseItem()
-            house_item['price'] = response.xpath("//p[@class = 'list__item__price__value']//text()").get()
-            house_item['price_per_meter'] = response.css('p.list__item__details__info.details--info--price::text').get()
-            house_item['url'] = response.css('a.list__item__content__title__name.link').attrib['href']
-            house_item['area'] = response.xpath("//p[@class='list__item__details__icons__element__desc']//text()").extract_first()
-            house_item['number_of_rooms'] = response.xpath('//ul/li[2]/div/p[2]//text()').extract_first()
+            
+            try:
+                
+                house_item['price'] = re.sub("[^0-9]", "", house.css('div.list__item__picture__price').get())
+                house_item['url'] = house.css('a.list__item__content__title__name.link').attrib['href']
+                house_item['numbers_of_rooms'] = re.sub("[^.0-9]", "", house.css('li.list__item__details__icons__element.details--icons--element--l_pokoi').get())
+                house_item['area'] = re.sub("[^.0-9]", "",house.css('p.list__item__details__icons__element__desc::text').get())
+                #house_item['price_per_meter'] = float(round((house_item['price']) / float(house_item['area'])), 2)
```

Comment:
To będziemy wyliczać w Pipelines (dalszym procesowaniu danych). A nie w tym miejscu.